controller/customdomain_utils.go

```diff
@@ -104,6 +118,82 @@ func FindCustomDomainCondition(conditions []customdomainv1alpha1.CustomDomainCon
 	return nil
 }
 
+// Take an ingress controller managed by the custom domains operator and release it back to the
+// cluster ingress operator. Also schedule it onto customer worker nodes from the Red Hat managed infra
+// nodes.
+func (r *CustomDomainReconciler) returnIngressToClusterIngressOperator(reqLogger logr.Logger, instance *customdomainv1alpha1.CustomDomain) (ctrl.Result, error) {
+	reqLogger.Info(fmt.Sprintf("Removing operator management labels from %s's underlying ingress controller", instance.Name))
+
+	ingressName := instance.Name
+	customIngress := &operatorv1.IngressController{}
+
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: ingressOperatorNamespace,
+		Name:      ingressName,
+	}, customIngress)
+	reqLogger.Info(fmt.Sprintf("Fetching ingress controller of name %s", instance.Name))
+
+	customIngress.Labels = map[string]string{}
+	if err != nil {
+		if kerr.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			reqLogger.Error(err, fmt.Sprintf("Ingresscontroller %s in %s namespace not found, or was never created", ingressName, ingressOperatorNamespace))
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	customIngress.Spec.NodePlacement = &operatorv1.NodePlacement{
+		NodeSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{"node-role.kubernetes.io/worker": ""},
+		},
+		Tolerations: []corev1.Toleration{},
+	}
+
+
+	reqLogger.Info(fmt.Sprintf("Updating ingress %s with new node placement on worker node, removing tolerations for infra nodes", instance.Name))
+	err = r.Client.Update(context.TODO(), customIngress)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Error updating ingresscontroller %s in %s namespace", ingressName, ingressOperatorNamespace))
+		return reconcile.Result{}, err
+	}
+
+
+	userSecret := &corev1.Secret{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: instance.Spec.Certificate.Namespace,
+		Name:      instance.Spec.Certificate.Name,
+	}, userSecret)
+	reqLogger.Info(fmt.Sprintf("Updating secret to remove custom domain labels from secret %s", userSecret.Name))
+
+	userSecret.Labels = map[string]string{}
```

Comment:
I think we'll need the same tweaks here as @mjlshen suggested [above](https://github.com/openshift/custom-domains-operator/pull/120/files#r1294122792)